### DISCUSSION TITLE
20230824 Weekly Development Meeting

### DISCUSSION THREAD

**Discussion by josecelano** [nautilus-cyberneering/.github#55] [Meetings]
Expected duration: 2 to 4 hours with breaks

Communicated attendants:

- CGarnham(@da2ce7)  - x
- JCelano(@josecelano) - x 
- CBosse(@cgbosse) - x
- MMelchor(@mario-nt) - x
- GByrne(@grmbyrn) - x

-------------------------------------------------------------

## Agenda:

1. Everyone will take their turn and:
- Go over his notes added as comments for the meeting which will have been added in advance to give an update of all the work done during the previous week indicating.
	- The issues worked on
	- The progress made on them
	- Other things done
- Inform of any faced challenges and request help if necessary to work on specific issues.
- Highlight the encountered difficulties, especially if there is a risk of them causing delays or affecting end product's quality.
- Communicate what will be the next week's tasks.
- List any new topics for further discussion in an additional Technical meeting that he noted for future discussion in comments to this discussion either before the meeting or after his turn so that they get considered.

During the meeting every team member will take their turns and follow this sequence.
The topics for further discussion will be noted in advance by each the team member and to the meetings agenda.

2. At the end of the meeting if there any new technical topics to be discussed further it will be decided to
set another meeting date for the discussion of these or to hold the meeting hereafter.

Also the persons to be part of the meeting will be decided on so that the others will be free to continue their own work.

**Comment by josecelano** [comment]
### Torrust

#### Torrust Index Backend

- https://github.com/torrust/torrust-index-backend/issues/255
- https://github.com/torrust/torrust-index-backend/issues/256
- https://github.com/torrust/torrust-index-backend/issues/257
- PR reviews

#### Academic Torrents Importer

A [Rust sample console app](https://github.com/torrust/academic-torrents-importer) to import torrents from https://academictorrents.com/ into Torrust Index.

It still needs to be finished and made public. I have to:

- [ ] Clean the code
- [ ] Add a README

The goals for this repo are:

- Test the Torrust Index with real torrents. In fact, I've found two bugs so far ([#255](https://github.com/torrust/torrust-index-backend/issues/255) and [#256](https://github.com/torrust/torrust-index-backend/issues/256))
- Provide a sample app to the community for people interested in migrating their Torrent Indexes from other solutions to ours.
- Extract the [backend API client in Rust](https://github.com/torrust/academic-torrents-importer/tree/main/src/target/api) from tests and test it so that in the future, we can convert it into a new repo/crate.

After finishing this example, I want to write two blog posts:

- How to migrate your torrents from your current Torrent Index to the Torrust Index.
- How to deploy the Torrust Index to a cloud provided (probably Digital Ocean) using real torrents from https://academictorrents.com/. I want to show how you can clone that site. It would be helpful for people who want to setup their own Index (whether they have a current solution with data or not).

### Nautilus Librarian

I was trying to update dependencies https://github.com/Nautilus-Cyberneering/nautilus-librarian/pulls. We have some security alerts.

I also found some broken things and opened new issues:

- https://github.com/Nautilus-Cyberneering/nautilus-librarian/issues/119
- https://github.com/Nautilus-Cyberneering/nautilus-librarian/issues/120
- https://github.com/Nautilus-Cyberneering/nautilus-librarian/issues/121
- https://github.com/Nautilus-Cyberneering/nautilus-librarian/issues/122

**Comment by mario-nt** [comment]
### Paperwork/Other

-   Setting up the new corporate email


### Torrust

-    Continue working on https://github.com/torrust/torrust-index-frontend/issues/205

- Writting a blog post for what a torrent tracker is.

- Researching about torrent files, trackers, etc. 

### Training

- Learning Rust
- Learning  Nuxt

**Comment by da2ce7** [comment]
### Work on Testing Workflows:
- https://github.com/torrust/torrust-tracker/pull/364
- https://github.com/torrust/torrust-index-backend/pull/246

### Work on Container Workflows:
- https://github.com/torrust/torrust-tracker/pull/363
- https://github.com/torrust/torrust-index-backend/pull/248

**Comment by grmbyrn** [comment]
## Admin
- Setting up email and signing documents.

## Svelte
- Finishing up a couple of projects to help with learning Svelte.
- Planning structure of the Git Secure Guide.
- Forking, cloning and introducing Svelte into Git Secure Guide repo through Vite.
- Working on creating the repo in Svelte.

## Blog
- Writing blogs as part of diary on my progress with Nautilus. Focusing on specific parts of Svelte as part of spaced repetition to encourage memory of these concepts. https://medium.com/@grmbrn89
